src/file.ts

```diff
@@ -2996,20 +2995,29 @@ class File extends ServiceObject<File> {
    */
 
   isPublic(callback?: IsPublicCallback): Promise<IsPublicResponse> | void {
-    gaxiosRequest({
-      method: 'HEAD',
-      url: `http://${
-        this.bucket.name
-      }.storage.googleapis.com/${encodeURIComponent(this.name)}`,
-    }).then(
-      () => callback!(null, true),
-      (err: GaxiosError) => {
-        if (err.code === '403') {
+    const callbackFunction = function (err: Error | ApiError | null) {
```

Comment:
From a style standpoint it doesn't appear we define functions in this manner anywhere else in this file (and others). I would suggest we either define it as an arrow function or move it inline to util.makeRequest as an arrow function.